### PR TITLE
Add initial Bazel support

### PR DESCRIPTION
**Pull request by fishy**
There's still a small bug upstream[0] which requires some manual
intervene so I didn't include it into CI yet, but it's already working
:)

[0] https://github.com/bazelbuild/bazel-gazelle/issues/724

### CODE REVIEW

File: secrets/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        "//batcherror:go_default_library",
+        "//filewatcher:go_default_library",
+        "//log:go_default_library",
+    ],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = [
+        "secrets_test.go",
```
Comment (konradreiche) [inline secrets/BUILD.bazel]: How does this work, I assume the list in here gets outgenerated?

Comment (fishy) [reply secrets/BUILD.bazel]: yes, that's auto generated by gazelle, by running:

```
bazel run //:gazelle
```

File: README.md [hunk 2]
```diff
@@ -24,6 +25,31 @@ We did not include `baseplate.thrift` file into this repo to avoid duplications.
 This directory will be regenerated when either thrift compiler or
 `baseplate.thrift` changed significantly.
 
+## Bazel support
+
+This project also comes with *optional* [Bazel][bazel] support.
+It's optional as in you can totally ignore Bazel and still use go toolchain,
+but the added support will make it eaiser for projects using Bazel to add this
```
Comment (pacejackson) [inline README.md]: `eaiser` => `easier`

File: README.md [hunk 3]
```diff
@@ -24,6 +25,31 @@ We did not include `baseplate.thrift` file into this repo to avoid duplications.
 This directory will be regenerated when either thrift compiler or
 `baseplate.thrift` changed significantly.
 
+## Bazel support
+
+This project also comes with *optional* [Bazel][bazel] support.
+It's optional as in you can totally ignore Bazel and still use go toolchain,
```
Comment (pacejackson) [inline README.md]: "still use `the` go toolchain"

File: README.md [hunk 4]
```diff
@@ -24,6 +25,31 @@ We did not include `baseplate.thrift` file into this repo to avoid duplications.
...
+but the added support will make it eaiser for projects using Bazel to add this
+project as a dependency.
+
+When you made a change to `go.mod` file,
+please run the following command to update the reflect the changes in Bazel:
```
Comment (pacejackson) [inline README.md]:
```suggestion
please run the following command to reflect the changes in Bazel:
```

**Pull request merged**